src/core/Akka.FSharp.Tests/RemoteSpecs.fs

```diff
@@ -0,0 +1,71 @@
+﻿module Akka.FSharp.Tests.RemoteSpecs
+
+open Akka.Actor
+open Xunit
+open Xunit.Abstractions
+open Akka.FSharp
+open Akka.Remote
+open Akka.TestKit
+open Serialization
+
+let remoteConfig port = 
+        $"""
+        akka {{ 
+            actor {{
+                ask-timeout = 5s
+                provider = remote
+            }}
+            remote {{
+                dot-netty.tcp {{
+                    port = %i{port}
+                    hostname = localhost
+                }}
+            }}
+        }}
+        """
+        |> Configuration.parse
+
+let getAddress (actorSystem:ActorSystem) =
+    let es = actorSystem :?> ExtendedActorSystem
+    es.Provider.DefaultAddress
+    
+type TestUnion = 
+    | A of string
+    | B of int * string
+
+type TestUnion2 = 
+    | C of string * TestUnion
+    | D of int
+
+type RemoteSpecs(output:ITestOutputHelper) as this =
+    inherit AkkaSpec((remoteConfig 0), output)
+    
+    do exprSerializationSupport this.Sys
+    
+    /// Retrieve the bound address used by Sys
+    member this.GetAddress =
+        getAddress this.Sys
+        
+    [<Fact>]
+    member _.``can serialize and deserialize discriminated unions over remote nodes using default serializer`` () =
+        
+        // arrange
+        use clientSys = System.create "clientSys" (remoteConfig 0)
+        let addr2 = getAddress clientSys
+        
+        // act
+        let aref = 
+            spawne clientSys "a-1" <@ actorOf2 (fun mailbox msg -> 
+                   match msg with
+                   | C("a-11", B(11, "a-12")) -> mailbox.Sender() <! msg
+                   | _ -> mailbox.Unhandled msg) @>
```

Comment:
We have a clear reproduction here - we receive a `JObject` instead of a DU here.

Reply:
Updated this to use Hyperion - even a basic Newtonsoft.Json serialization setup can't handle DUs

Reply:
If I remember correctly - but I might be wrong - the issue might be caused by the fact that you didn't configured JSON.NET to include type name as part of the payload. Since at the deserialization time payload type is still unknown and JSON.NET will use JObject as representation of `System.Object` instance.

Reply:
Yes, with type hints Newtonsoft.Json correctly deserializes DUs: 

https://github.com/akkadotnet/akka.net/blob/d2014528380ac7c00ea97084ba59495ba8ae76f2/src/core/Akka.FSharp.Tests/SerializationSpecs.fs#L41-L50

We disabled exposing type hints during the advent of Akka.NET v1.3 because of `mscorlib` namespace issues introduced by the creation of .NET Core: https://twitter.com/Danthar/status/1426204694310014981 cc @Danthar 

We could fix the issue by turning type hints on, but it would blow up anyone doing .NET Framework --> .NET Core interop with Newtonsoft.Json (population of people doing that is great than zero) - or we could make it configurable to turn that setting on or off.

Reply:
I think you can potentially handle that as special case inside of typed actor itself: https://github.com/akkadotnet/akka.net/blob/d2014528380ac7c00ea97084ba59495ba8ae76f2/src/core/Akka.FSharp/FsApi.fs#L291-L297

Before down casting message, check if `obj` is `JObject` and if so use its [ToObject](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Linq_JToken_ToObject__1_1.htm) method.